PWGHF/TableProducer/treeCreatorToXiPi.cxx

```diff
@@ -164,26 +172,67 @@ DECLARE_SOA_TABLE(HfToXiPiFulls, "AOD", "HFTOXIPIFULL",
                   full::FlagMcMatchRec, full::DebugMcRec, full::OriginRec, full::CollisionMatched);
 
+DECLARE_SOA_TABLE(HfToXiPiLite, "AOD", "HFTOXIPILITE",
+                  full::XPv, full::YPv, full::ZPv, collision::NumContrib, collision::Chi2,
+                  full::XDecayVtxCharmBaryon, full::YDecayVtxCharmBaryon, full::ZDecayVtxCharmBaryon,
+                  full::XDecayVtxCascade, full::YDecayVtxCascade, full::ZDecayVtxCascade,
+                  full::XDecayVtxV0, full::YDecayVtxV0, full::ZDecayVtxV0,
+                  full::SignDecay,
+                  full::PxCharmBaryon, full::PyCharmBaryon, full::PzCharmBaryon,
+                  full::PxPiFromCharmBaryon, full::PyPiFromCharmBaryon, full::PzPiFromCharmBaryon,
+                  full::PxPiFromCasc, full::PyPiFromCasc, full::PzPiFromCasc,
+                  full::PxPosV0Dau, full::PyPosV0Dau, full::PzPosV0Dau,
+                  full::PxNegV0Dau, full::PyNegV0Dau, full::PzNegV0Dau,
+                  full::ImpactParCascXY, full::ImpactParPiFromCharmBaryonXY,
+                  full::ErrImpactParCascXY, full::ErrImpactParPiFromCharmBaryonXY,
+                  full::InvMassLambda, full::InvMassCascade, full::InvMassCharmBaryon,
+                  full::EtaV0PosDau, full::EtaV0NegDau, full::EtaPiFromCasc, full::EtaPiFromCharmBaryon,
+                  full::DcaXYToPvV0Dau0, full::DcaXYToPvV0Dau1, full::DcaXYToPvCascDau,
+                  full::DcaCascDau, full::DcaV0Dau, full::DcaCharmBaryonDau,
+                  full::ErrorDecayLengthCharmBaryon, full::NormImpParCascade, full::NormImpParPiFromCharmBar,
+                  full::IsPionGlbTrkWoDca, full::PionItsNCls,
+                  full::PidTpcInfoStored, full::PidTofInfoStored,
+                  full::TpcNSigmaPiFromCharmBaryon, full::TpcNSigmaPiFromCasc, full::TpcNSigmaPiFromLambda, full::TpcNSigmaPrFromLambda,
+                  full::TofNSigmaPiFromCharmBaryon, full::TofNSigmaPiFromCasc, full::TofNSigmaPiFromLambda, full::TofNSigmaPrFromLambda,
+                  full::FlagMcMatchRec, full::OriginRec, full::CollisionMatched);
+
 } // namespace o2::aod
 
 /// Writes the full information in an output TTree
 struct HfTreeCreatorToXiPi {
 
   Produces<o2::aod::HfToXiPiFulls> rowCandidateFull;
+  Produces<o2::aod::HfToXiPiLite> rowCandidateLite;
+  Produces<o2::aod::HfToXiPiEv> rowEv;
 
-  using MyTrackTable = soa::Join<aod::Tracks, aod::TrackSelection>;
+  Configurable<float> zPvCut{"zPvCut", 10., "Cut on absolute value of primary vertex z coordinate"};
+
+  using MyTrackTable = soa::Join<aod::Tracks, aod::TrackSelection, aod::TracksExtra>;
+  using MyEventTable = soa::Join<aod::Collisions, aod::EvSels>;
 
   void init(InitContext const&)
   {
   }
 
+  template <class TMyEvents, typename T>
+  void fillEvent(const T& collision, float cutZPv, const bool& newEvent, bool& isEvSel8, bool& isEventZSel)
```

Comment:
```suggestion
  void fillEvent(const T& collision, float cutZPv, bool newEvent, bool& isEvSel8, bool& isEventZSel)
```

Reply:
Why do you pass `isEvSel8` and `isEventZSel` by reference? You reset them to `false` for each collision and I don't see the changed values used anywhere outside the function.

Reply:
Can you not just directly fill:
```c++
rowEv(newEvent, collision.sel8(), std::abs(collision.posZ()) < cutZPv);
```
?

Reply:
Yes, sure, I modified it according to this suggestion